pkg/pipeline/ingest/ingest_kafka.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	 http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package ingest
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/netobserv/flowlogs2metrics/pkg/api"
+	"github.com/netobserv/flowlogs2metrics/pkg/config"
+	kafka "github.com/segmentio/kafka-go"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+type ingestKafka struct {
+	kafkaParams api.IngestKafka
+	kafkaReader *kafka.Reader
+}
+
+// Ingest ingests entries from kafka topic and sends them down the pipeline
+func (r *ingestKafka) Ingest(process ProcessFunction) {
+	lines := make([]interface{}, 1)
+
+	for {
+		m, err := r.kafkaReader.ReadMessage(context.Background())
+		if err != nil {
+			log.Errorln(err)
+		}
+		log.Debugf("message at topic:%v partition:%v offset:%v	%s = %s\n", m.Topic, m.Partition, m.Offset, string(m.Key), string(m.Value))
+		lines[0] = string(m.Value)
+		process(lines)
+	}
+}
+
+// NewIngestKafka create a new ingester
+func NewIngestKafka() (Ingester, error) {
+	log.Debugf("entering NewIngestKafka")
+	ingestKafkaString := config.Opt.PipeLine.Ingest.Kafka
+	log.Debugf("ingestKafkaString = %s", ingestKafkaString)
+	var jsonIngestKafka api.IngestKafka
+	err := json.Unmarshal([]byte(ingestKafkaString), &jsonIngestKafka)
+	if err != nil {
+		return nil, err
+	}
+
+	// connect to the kafka server
+	brokers := jsonIngestKafka.Brokers
+	topic := jsonIngestKafka.Topic
+	groupId := jsonIngestKafka.GroupId
+	kafkaReader := kafka.NewReader(kafka.ReaderConfig{
+		Brokers: brokers,
+		Topic:   topic,
+		GroupID: groupId,
+	})
```

Comment:
I think you should at least add `StartOffset` and `GroupBalancers` that are useful

Reply:
I think that `StartOffset` should always be the default `FirstOffset` value. Any use case to ignore uncommited messages?

Reply:
Well if you stopped reading for some time and have a lot of old messages you don't care about you may want to skip them to directly start at new elements ? Or you need to flush kafka before restart ?

I was thinking at the situations where an admin wants to capture a specific period of time. He would not expect messages before he pressed "start".

But I agree by default if would be FirstOffset :+1:

Reply:
I understand. Thanks for the clarification!

Reply:
@jpinsonneau @mariomac What do we need in terms of authentication/authorization to use the kafka topic?

Reply:
You first need to create a kafka user and [set credentials in the dialer](https://github.com/segmentio/kafka-go#sasl-support)

For now I only played with [strimzi](https://strimzi.io/) default config which is:

```yaml
kind: KafkaUser
apiVersion: kafka.strimzi.io/v1beta2
metadata:
  name: my-user
  labels:
    strimzi.io/cluster: my-cluster
  namespace: openshift-operators
spec:
  authentication:
    type: tls
  authorization:
    type: simple
    acls:
      - resource:
          type: topic
          name: my-topic
          patternType: literal
        operation: Read
        host: '*'
      - resource:
          type: topic
          name: my-topic
          patternType: literal
        operation: Describe
        host: '*'
      - resource:
          type: group
          name: my-group
          patternType: literal
        operation: Read
        host: '*'
      - resource:
          type: topic
          name: my-topic
          patternType: literal
        operation: Write
        host: '*'
      - resource:
          type: topic
          name: my-topic
          patternType: literal
        operation: Create
        host: '*'
      - resource:
          type: topic
          name: my-topic
          patternType: literal
        operation: Describe
        host: '*'
```

You should not need `Write` and `Create` authorization for reading.

How do you currently create your kafka instance ?

Reply:
I currently run kafka directly on my development VM. No user management.

Reply:
👍